### ISSUE TITLE
`jb update foo` should print an error when it doesn't do anything

### ISSUE THREAD

**Issue by tomwilkie** [jsonnet-bundler/jsonnet-bundler#121]
Took me three stabs just now:

```
twilkie@MacBook-Pro ksonnet % jb update github.com/grafana/cortex-jsonnets/cortex                      
twilkie@MacBook-Pro ksonnet % jb update github.com/grafana/cortex-jsonnets/cortex-mixin
twilkie@MacBook-Pro ksonnet % jb update github.com/grafana/cortex-jsonnet/cortex-mixin 
GET https://github.com/grafana/cortex-jsonnet/archive/1d69621ce3f7a703ddf56f5d9fb93faf4a3790bb.tar.gz 200
twilkie@MacBook-Pro ksonnet % 
````

**Comment by ghostsquad** [comment]
I prefer idempotent commands actually. It avoids needing to write stuff like `jb update <repo> || true` in automation.

At the end of the day, you are simply asking for the desired state to be "at the latest release". So if the actual state matches desired state, the exit code should indicate success, even if nothing was done.

**Comment by sh0rez** [comment]
> So if the actual state matches desired state, the exit code should indicate success, even if nothing was done.

Fair point in the case something is already up to date. I think it still wouldn't harm printing some "already up to date" message then.

More relevant is the case `jb update github.com/grafana/cortex-jsonnets/cortex-mixin`, which is straight up invalid, as the repository does not exist at all and is also unlikely part of the existing `jsonnetfile.json`.

I think it would be fair to fail if trying to update something that wasn't installed previously
